### PR TITLE
Missing vector tinting

### LINKED ISSUE #4003: The color of this arrow is the Element theme, could this be added to the color config?
### Steps to reproduce

Have a lot of unread messages, then go to the chat

image

### What happened?
color is not per theme config

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Have you submitted a rageshake?

No

### PR DESCRIPTION
**Pull request by ouchadam**
Applies the `colorPrimary` instead of a hardcoded element green #0DBD8B, fixes #4003

- Extracts some Context/View extensions to make tinting easier
- Replaces the `ic_voice_mic_recording.xml` vector background path with a dedicated background drawable 
- Uses #FF0000 as the placeholder colour as it makes it very obvious when a drawable hasn't been tinted
- Removes unused `ic_selected_community.xml` resource

* Uses a colorPrimary of #0000FF for the screenshots

| BEFORE JUMP | AFTER JUMP |
| --- | --- |
before-jump-to-unread|after-jump-to-unread

| BEFORE RADIO | AFTER RADIO |
| --- | --- | 
before-radio|after-radio

| BEFORE VOICE | AFTER VOICE |
| --- | --- |
before-voice|after-voice

### CODE REVIEW

File: vector/src/main/java/im/vector/app/core/extensions/ViewExtensions.kt [hunk 1]
```diff
@@ -58,3 +63,21 @@ fun ImageView.setDrawableOrHide(drawableRes: Drawable?) {
...
+    val drawable = ContextCompat.getDrawable(context, drawableRes)!!
+    DrawableCompat.setTint(drawable, context.fetchThemeColor(tint))
+    setImageDrawable(drawable)
+}
+
+fun View.setAttributeBackground(@AttrRes attributeId: Int) {
+    val typedValue = TypedValue()
+    context.theme.resolveAttribute(attributeId, typedValue, true)
+    setBackgroundResource(typedValue.resourceId)
+}
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/extensions/ViewExtensions.kt:83]: Have you seen the object ThemeUtils? It's not perfect, but there is some similarity with what is done here, maybe we should harmonize the code later.

File: vector/src/main/res/drawable/circle_with_halo.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            <solid android:color="#33FF0000" />
+        </shape>
+    </item>
+    <item
+        android:bottom="4dp"
+        android:left="4dp"
+        android:right="4dp"
+        android:top="4dp">
+        <shape android:shape="oval">
+            <solid android:color="#FF0000" />
```
Comment (bmarty) [inline vector/src/main/res/drawable/circle_with_halo.xml:14]: Thanks for using Red color (I do it too) to ensure that the tint is correctly applied at runtime.

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

**Review by bmarty — COMMENTED** [review]
Please see my remarks

File: vector/src/main/java/im/vector/app/core/extensions/ViewExtensions.kt (hunk 1 above)
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/core/extensions/ViewExtensions.kt:83]: ahhh, I'll update the extensions to call into the ThemeUtils

**Comment by ouchadam** [comment]
> What about those occurences?
> 
> * https://github.com/vector-im/element-android/blob/main/vector/src/main/res/drawable/ic_launcher_background.xml#L8
> * https://github.com/vector-im/element-android/blob/main/vector/src/main/res/drawable/ic_selected_community.xml#L8

unfortunately it doesn't seem like it's possible to use theme attributes in a `adaptive-icon` 

1.
```xml
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@android:color/holo_red_light" />
    <foreground android:drawable="@drawable/ic_launcher_foreground" />
</adaptive-icon>
```
2.
```xml
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="?colorPrimary" />
    <foreground android:drawable="@drawable/ic_launcher_foreground" />
</adaptive-icon>
```

3.
```xml
<background android:drawable="@drawable/ic_launcher_background" />

// ic_launcher_background.xml colorPrimary set to blue #0000FF
<vector xmlns:android="http://schemas.android.com/apk/res/android"
    android:width="108dp"
    android:height="108dp"
    android:viewportWidth="108"
    android:viewportHeight="108">
  <path
      android:pathData="m0,0h108v108h-108z"
      android:fillColor="?colorPrimary" />
</vector>
```


| 1 | 2 | 3 |
| --- | --- | --- |
|2021-09-28T15:31:50,633736941+01:00|2021-09-28T15:32:33,460967000+01:00|2021-09-28T15:36:06,762067490+01:00



for `ic_selected_community.xml`, I'll delete the file, doesn't seem to be used

**Comment by bmarty** [comment]
Sorry, I should have double check. No need to update https://github.com/vector-im/element-android/blob/main/vector/src/main/res/drawable/ic_launcher_background.xml, I though it was the background of the app at launch time, I read it too fast :)

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Pull request merged**